<launch>
    <arg name="fixed_frame" default="base" />
    <arg name="publish_in_fixed_frame" default="false" />
    <arg name="cloud_topic" default="/cloud_coloring/semantic_cloud" />
    <arg name="fuse_only" default="true" />
    <arg name="side_length" default="30" />
    <arg name="voxel_length" default="0.10" />
    <arg name="no_raytracing" default="false" />
    <arg name="fuse_semantic_window" default="true" />
    <arg name="semantic_window_size" default="10" />
    <arg name="remove_older" default="false" />
    <arg name="store_all" default="true" />
    <arg name="n_classes" default="32" />
    <arg name="node_prefix" default=""/>
    
    <node name="$(arg node_prefix)cloud_fusor_node" pkg="jetson_semantic_map" type="jetson_semantic_map_node" output="screen">
        <param name="fuse_semantic_window" value="$(arg fuse_semantic_window)" />
        <param name="remove_older" value="$(arg remove_older)" />
        <param name="store_all" value="$(arg store_all)" />
        <param name="fuse_only" value="$(arg fuse_only)" />
        <param name="side_length" value="$(arg side_length)" />
        <param name="voxel_length" value="$(arg voxel_length)" />
        <param name="semantic_window_size" value="$(arg semantic_window_size)" />
        <param name="no_raytracing" value="$(arg no_raytracing)" />
        <param name="fixed_frame" value="$(arg fixed_frame)" />
        <param name="publish_in_fixed_frame" value="$(arg publish_in_fixed_frame)" />
        <param name="n_classes" value="$(arg n_classes)" />
        <param name="cloud_topic" value="$(arg cloud_topic)" />
    </node>
</launch>
